examples/common-switches.mk

```diff
@@ -50,6 +50,7 @@ DIAGNOSTIC                ?= 0
 DISABLE_DOC               ?= 0
 DISABLE_TOOLS             ?= 0
 DNS_CLIENT                ?= 0
+DNSSD_SERVER              ?= 0
```

Comment:
add introduction on this feature switch in https://github.com/openthread/openthread/blob/master/examples/README.md

_@wgtdkp `BORDER_ROUTING `should be added as well._